experimental/framework/fluid-static/src/containerCode.ts

```diff
@@ -9,59 +9,152 @@ import {
+ */
+export class RootDataObject
+    // eslint-disable-next-line @typescript-eslint/ban-types
+    extends DataObject<{}, RootDataObjectProps, IFluidContainerEvents>
+    implements FluidContainer {
+    private readonly connectedHandler = (id: string) => this.emit("connected", id);
+    private readonly initialObjectsDirKey = "initial-objects-key";
+    private readonly _initialObjects: LoadableObjectRecord = {};
+
+    private get initialObjectsDir() {
+        const dir = this.root.getSubDirectory(this.initialObjectsDirKey);
+        if (dir === undefined) {
+            throw new Error("InitialObjects sub-directory was not initialized");
+        }
+        return dir;
+    }
+
     protected async initializingFirstTime(props: RootDataObjectProps) {
+        this.root.createSubDirectory(this.initialObjectsDirKey);
+
         // Create initial objects provided by the developer
         const initialObjectsP: Promise<void>[] = [];
-        Object.entries(props.initialObjects).forEach(([id, dataObjectClass]) => {
+        Object.entries(props.initialObjects).forEach(([id, objectClass]) => {
             const createObject = async () => {
-                const obj = await this.createInternal(dataObjectClass);
-                this.root.set(id, obj.handle);
+                const obj = await this.create(objectClass);
+                this.initialObjectsDir.set(id, obj.handle);
             };
             initialObjectsP.push(createObject());
         });
 
         await Promise.all(initialObjectsP);
     }
 
-    protected async hasInitialized() { }
+    protected async hasInitialized() {
+        this.runtime.on("connected", this.connectedHandler);
+
+        // We will always load the initial objects so they are avaiable to the developer
+        const loadInitialObjectsP: Promise<void>[] = [];
+        for (const [key, value] of Array.from(this.initialObjectsDir.entries())) {
+            const loadDir = async () => {
+                const obj = await value.get();
+                Object.assign(this._initialObjects, { [key]: obj });
+            };
+            loadInitialObjectsP.push(loadDir());
+        }
 
-    public async createDataObject<T extends IFluidLoadable>(
-        dataObjectClass: IFluidStaticDataObjectClass,
-        id: string,
-    ) {
-        const obj = await this.createInternal(dataObjectClass);
-        this.root.set(id, obj.handle);
-        return obj;
+        await Promise.all(loadInitialObjectsP);
+    }
+
+    public dispose() {
+        // remove our listeners and continue disposing
+        this.runtime.off("connected", this.connectedHandler);
+        super.dispose();
+    }
+
+    public get audience(): IAudience {
+        return this.context.getAudience();
+    }
+
+    public get clientId() {
+        return this.context.clientId;
+    }
+
+    public get initialObjects(): LoadableObjectRecord {
+        if (Object.keys(this._initialObjects).length === 0) {
```

Comment:
Would it be invalid to have no initial objects but have a few dynamic objects? Or is that not considered for rootdataobject?

Reply:
Dynamic objects needs to be stored on an initialObject (or another dynamic object) with out an initialObject you cannot create and store dynamic objects.